### PR TITLE
Issue #6320: added test cases from regression for missing mutation

### PR DESCRIPTION
**Pull request by rnveach**
Issue #6320

This is for coding package only.

### CODE REVIEW

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/InputIllegalInstantiationMap.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
+IllegalInstantiation
+classes = java.util.Map, java.util.HashMap
+tokens = (default)CLASS_DEF
+
+
+*/
+
+//non-compiled syntax: the import is missing and is used to test the code
```
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/InputIllegalInstantiationMap.java:9]: It is our limitation, we do not operate on bad file at all.

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/InputIllegalInstantiationMap.java (hunk 1 above)
Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/InputIllegalInstantiationMap.java:9]: We keep only very bad files.
A bit wrong is not even considered.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral.java [hunk 2]
```diff
@@ -1071,4 +1071,44 @@ private void n() {
         New1.field = (int)a;
     }
 
+    private void o() {
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral.java:1074]: Better to keep file size in reasonable size  to not scroll to much to top to see config.
Can we create new file for such code ?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisValidateOnlyOverlappingTrue.java [hunk 3]
```diff
@@ -475,3 +475,20 @@ void foo55(String methodParam) {
 
     boolean foo57(boolean booleanField) { booleanField = !booleanField;  return booleanField; }
 }
+class TestClass2 {
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisValidateOnlyOverlappingTrue.java:478]: Better to move to new file if possible

**Review by romani — CHANGES_REQUESTED** [review]
It might be good point to discuss, what size of Input file we should consider to max ?  Is this quantifiable ?

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/InputIllegalInstantiationMap.java (hunk 1 above)
Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/InputIllegalInstantiationMap.java:9]: Can you forbid class that do not need import ?

Comment (rnveach) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegalinstantiation/InputIllegalInstantiationMap.java:9]: Test case removed.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral.java (hunk 2 above)
Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral.java:1074]: I recommend we see if we can enforce something through CI. Sounds like a good GSoC simple issue.

Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral.java:1074]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisValidateOnlyOverlappingTrue.java (hunk 3 above)
Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisValidateOnlyOverlappingTrue.java:478]: Done.

**Comment by rnveach** [comment]
@romani All done.

**Review by romani — APPROVED** [review]

**Pull request merged**